setup.py

```diff
@@ -16,11 +16,19 @@ def readme():
         'Development Status :: 3 - Alpha',
         'License :: OSI Approved :: MIT License',
         'Natural Language :: English',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.6',
         'Intended Audience :: Developers',
         ],
       py_modules = ['sqla_yaml_fixtures'],
-      install_requires = ['SQLAlchemy', 'PyYAML'],
+      install_requires = [
+          'SQLAlchemy',
+          'PyYAML'
+      ],
+      extras_require={
+          ':python_version=="2.7"': ['backports.functools_lru_cache'],
```

Comment:
Ok, I propose this:

```
from __future__ import print_function
from setuptools import setup
import sys

# Assume pip is new enough
pip_message = None
try:
    import pip
    pip_version = tuple([int(x) for x in pip.__version__.split('.')[:3]])
    if pip_version < (9, 0, 1) :
        pip_message = 'Your pip version is out of date, please install pip >= 9.0.1. '\
        'pip {} detected.'.format(pip.__version__)
        
except Exception:
    pip_message = 'This may be due to an out of date pip. Make sure you have pip >= 9.0.1.'


if pip_message:
    # Some error from pip version check
    error = """
    Python {py} detected.
    Exiting setup: 
    {pip}
    """.format(py=sys.version_info, pip=pip_message )

    print(error, file=sys.stderr)
    sys.exit(1)
```